add support for plots x dict

dberenbaum opened the pull request:
Leftover from https://github.com/iterative/dvc/pull/8421.

@skshetry I had to manually run `python gen.py > schema.json` to get the pre-commit to pass.

skshetry commented on examples/valid/plots_list.dvc.yaml:
```diff
@@ -20,8 +20,20 @@ plots:
+        precision_recall.json: precision
+        roc.json: tpr
   - evaluation/importance.png: # Plot an image as dict.
-  - evaluation/importance.png # Plot an image as str.
```
Let's keep this. If you don't mind, it would be nice to change to a different path.

dberenbaum replied:
Sorry, mistakenly removed.

skshetry commented on examples/valid/plots_list.dvc.yaml:
```diff
@@ -20,8 +20,20 @@ plots:
+        dir/actual.csv: actual
+      y:
+        dir/preds.csv: predicted
+      template: confusion - evaluation/importance.png # Plot an image as str.
```
Is this legal? Or, is it just an example?

dberenbaum replied:
Looks like I just messed up editing the file.

skshetry commented:
> @skshetry I had to manually run `python gen.py > schema.json` to get the pre-commit to pass.

It should be auto-generated by `pre-commit` (and if `schema.json` gets modified, pre-commit will fail). so, the workflow now is to:
1. change `gen.py` and git add.
2. do `git commit` or just `pre-commit`, and it will generate `schema.json`. If it gets modified, pre-commit will fail. Add the modified schema.json and retry.


I did this so that now you can just change `gen.py` from github.dev and pre-commit.ci will autogenerate `schema.json` on your PR. So you don't need to checkout locally.

skshetry commented:
> @skshetry I had to manually run `python gen.py > schema.json` to get the pre-commit to pass.

It should be auto-generated by `pre-commit` (and if `schema.json` gets modified, pre-commit will fail). so, the workflow now is to:
1. change `gen.py` and git add.
2. do `git commit` or just `pre-commit`, and it will generate `schema.json`. If it gets modified, pre-commit will fail. Add the modified schema.json and retry.


I did this so that now you can just change `gen.py` from github.dev and pre-commit.ci will autogenerate `schema.json` on your PR. So you don't need to checkout locally.

dberenbaum commented:
I'm not finding that `pre-commit` is writing the output of `gen.py` to `schema.json`:

```
$ git status
On branch plots-x-dict
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   gen.py

(dvc) dave@davids-air:~/Code/dvcyaml-schema [plots-x-dict] 11:41:09
$ pre-commit
zsh: /Users/dave/miniforge3/bin/pre-commit: bad interpreter: /Users/dave/miniforge3/bin/python3.9: no such file or directory
[WARNING] Unexpected key(s) present on https://github.com/pre-commit/mirrors-mypy: additional_dependencies
black....................................................................Passed
check for added large files..............................................Passed
check for case conflicts.................................................Passed
check docstring is first.................................................Passed
check that executables have shebangs.....................................Passed
check json...........................................(no files to check)Skipped
check for merge conflicts................................................Passed
check toml...........................................(no files to check)Skipped
check yaml...........................................(no files to check)Skipped
debug statements (python)................................................Passed
fix end of files.........................................................Passed
mixed line ending........................................................Passed
sort simple yaml files...............................(no files to check)Skipped
trim trailing whitespace.................................................Passed
codespell................................................................Passed
pyupgrade................................................................Passed
isort....................................................................Passed
flake8...................................................................Passed
pylint...................................................................Passed
mypy.....................................................................Passed
generate-schema......................................(no files to check)Skipped
- hook id: gen
test-schema..............................................................Passed
- hook id: test
- duration: 0.46s

============================= test session starts ==============================
platform darwin -- Python 3.10.2, pytest-7.2.0, pluggy-1.0.0 -- /Users/dave/.cache/pre-commit/repompyr3d59/py_env-python3/bin/python
cachedir: .pytest_cache
rootdir: /Users/dave/Code/dvcyaml-schema, configfile: pyproject.toml, testpaths: tests.py
collected 16 items

tests.py::test_valid_examples[plots_list] PASSED                         [  6%]
tests.py::test_valid_examples[foreach] PASSED                            [ 12%]
tests.py::test_valid_examples[plots_dict] PASSED                         [ 18%]
tests.py::test_valid_examples[metrics] PASSED                            [ 25%]
tests.py::test_valid_examples[minimal] PASSED                            [ 31%]
tests.py::test_valid_examples[outs] PASSED                               [ 37%]
tests.py::test_valid_examples[vars] PASSED                               [ 43%]
tests.py::test_valid_examples[params] PASSED                             [ 50%]
tests.py::test_valid_examples[simple] PASSED                             [ 56%]
tests.py::test_invalid_examples[toplevel.extra] PASSED                   [ 62%]
tests.py::test_invalid_examples[plots.extra] PASSED                      [ 68%]
tests.py::test_invalid_examples[outs.extra] PASSED                       [ 75%]
tests.py::test_invalid_examples[foreach.extra] PASSED                    [ 81%]
tests.py::test_invalid_examples[metrics.extra] PASSED                    [ 87%]
tests.py::test_invalid_examples[live.extra] PASSED                       [ 93%]
tests.py::test_invalid_examples[stage.extra] PASSED                      [100%]

============================== 16 passed in 0.33s ==============================

(dvc) dave@davids-air:~/Code/dvcyaml-schema [plots-x-dict] 11:41:20
$ git status
On branch plots-x-dict
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   gen.py
```
